src/com/sap/piper/integration/TransportManagementService.groovy

```diff
@@ -60,8 +66,10 @@ class TransportManagementService implements Serializable {
             echo("URL: '${url}', File: '${file}'")
         }
 
+        def proxy = config.proxy ? config.proxy : script.env.HTTP_PROXY
+
         script.sh """#!/bin/sh -e
-                curl -H 'Authorization: Bearer ${token}' -F 'file=@${file}' -F 'namedUser=${namedUser}' -o responseFileUpload.txt  --fail '${url}/v2/files/upload'
+                curl ${proxy ? '--proxy ' + proxy + ' ' : ''} -H 'Authorization: Bearer ${token}' -F 'file=@${file}' -F 'namedUser=${namedUser}' -o responseFileUpload.txt  --fail '${url}/v2/files/upload'
```

Comment:
[clarification] this is OK like this. `curl` itself takes environment variable `http_proxy` into account, but the proxy might come from the piper configuration (... and not from the environment variable)